File: ruby/lib/enolib/elements/missing/missing_section_element.rb
# frozen_string_literal: true

module Enolib
  class MissingSectionElement < MissingElementBase
    def to_s
      if @key
        "#<Enolib::MissingSectionElement key=#{@key}>"
      else
        '#<Enolib::MissingSectionElement>'
      end
    end
  end
end

File: javascript/lib/elements/missing/missing_section_element.js
import { MissingElementBase } from './missing_element_base.js';

export class MissingSectionElement extends MissingElementBase {
    get [Symbol.toStringTag]() {
        return 'MissingSectionElement';
    }
    
    toString() {
        if(this._key === null)
            return `[object MissingSectionElement]`;
        
        return `[object MissingSectionElement key=${this._key}]`;
    }
}
